docs(tem): quickstart and how-tos (#942)
scaleway/docs-content: managed-services/transactional-email/how-to/generate-api-keys-for-api-and-smtp-sending.mdx

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/).
  - You have a domain name.
  - You have [configured your domain with Transactional Email](/managed-services/transactional-email/how-to/configure-domain-with-transactional-email).
</Message>

## Generate an API key and send emails via the API

To send your emails via the API, a valid API key and your Project's ID are required.

1. From the [Scaleway console](https://console.scaleway.com), navigate to the [Credentials](/console/my-account/concepts#credentials) tab of your Project Dashboard:
        <Lightbox src="scaleway-credentials-nav.webp" alt="" />
2. Scroll down to the **API Key** section:
        <Lightbox src="scaleway-api-keys.webp" alt="" />
3. Click **Generate new API key**. The following pop-up appears:
        <Lightbox src="scaleway-api-key-purpose.webp" alt="" />
4. Add an **API Key purpose** (for your own reference), or leave this blank. Then click **Generate API Key**.

>     The two parts of your API Key (access key and secret key) are displayed. Take careful note of the secret key, as it will not be recoverable or viewable after this point.

Reviewer: ```suggestion
    The two parts of your API Key (access key and secret key) are displayed. Note the secret key carefully, as it will not be recoverable or viewable after this point.
```

---
meta:
  title: How to generate API keys for API and SMTP sending
  description: This page explains how to generate API keys for API and SMTP sending
content:
  h1: How to generate API keys for API and SMTP sending
  paragraph: This page explains how to generate API keys for API and SMTP sending
dates:
  validation: 2022-11-07
  posted: 2022-11-07
---

This page shows you how to generate [API keys](/console/my-account/concepts/#api-key) to send [transactional emails](/managed-services/transactional-email/concepts/#transactional-email) via API and SMTP.

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/).
  - You have a domain name.
  - You have [configured your domain with Transactional Email](/managed-services/transactional-email/how-to/configure-domain-with-transactional-email).
  - You have added [SPF, DKIM](/managed-services/transactional-email/how-to/add-spf-dkim-records-to-your-domain) and [MX records](/managed-services/transactional-email/how-to/add-mx-records-to-your-domain) to your domain.
</Message>

## Generate an API key and send emails via the API

> To send your emails via the API, a valid API key and your Project's ID are required.

Reviewer 2: ```suggestion
To send your emails via the API, a valid API key and your Project ID are required.
```

        }
    ]
}
EOF

curl -X POST "https://api.scaleway.com/transactional-email/v1alpha1/regions/$REGION/emails" \
-H "X-Auth-Token: $TOKEN" \
-d @mail.json
```

<Message type="important">
 - If you have not configured your environment in your local machine, make sure you replace `\<project_id\>` with your Project ID, `$REGION` with `fr-par` and `$TOKEN` with your secret key.
 - Make sure you replace the email addresses with yours and your recipient's
</Message>

The following output should display:

```
{"emails":[{"id":"655c27f2-b2a3-4a9f-8e1f-3e6dc268b1c4","message_id":"3d928e21-187a-4539-b303-403156e37911","project_id":"3e44a89e-1f75-45f5-9b93-422db94165eb","mail_from":"alice@my.domain.example.com","rcpt_to":"bob@caramail.fr","rcpt_type":"to","created_at":"2022-08-01T07:55:36.758671147Z","updated_at":"2022-08-01T07:55:36.758671147Z","status":"new","status_details":"not yet processed","try_count":0,"last_tries":[]}]}%
```

## Generate an API key to send emails via SMTP

The Transactional Email SMTP relay is available at `smtp.tem.scw.cloud` on ports `25`, `587`, `2587`, `465` (TLS) and `2465` (TLS).

> To send your emails using SMTP, a valid API key and your Project's ID are required.

Reviewer 2: ```suggestion
To send your emails using SMTP, a valid API key and your Project ID are required.
```
